de-lib/src/main/java/org/iplantc/de/apps/widgets/client/view/editors/AppStepResourcesViewImpl.java

```diff
@@ -0,0 +1,169 @@
+
+import com.sencha.gxt.cell.core.client.form.ComboBoxCell;
+import com.sencha.gxt.widget.core.client.ContentPanel;
+import com.sencha.gxt.widget.core.client.form.SimpleComboBox;
+
+/**
+ * @author psarando
+ */
+public class AppStepResourcesViewImpl extends ContentPanel implements AppStepResourcesView, Editor<AppTemplateStepRequirements> {
+
+    private static final long ONE_GB = 1024 * 1024 * 1024;
+
+    @Path("minCPUCores")
+    @UiField (provided = true)
+    SimpleComboBox<Double> minCPUCoresEditor;
+
+    @Path("minMemoryLimit")
+    @UiField (provided = true)
+    SimpleComboBox<Long> memory;
+
+    @Path("minDiskSpace")
+    @UiField (provided = true)
+    SimpleComboBox<Long> minDiskSpaceEditor;
+
+    @UiField(provided = true)
+    AppTemplateWizardAppearance appearance;
+
+    interface EditorDriver extends SimpleBeanEditorDriver<AppTemplateStepRequirements, AppStepResourcesViewImpl> {
+    }
+
+    private final AppStepResourcesViewImpl.EditorDriver editorDriver =
+            GWT.create(AppStepResourcesViewImpl.EditorDriver.class);
+
+
+    @UiTemplate("AppStepResourcesView.ui.xml")
+    interface AppStepResourcesViewUiBinder extends UiBinder<Widget, AppStepResourcesViewImpl> {
+
+    }
+
+    private static final AppStepResourcesViewUiBinder uiBinder = GWT.create(AppStepResourcesViewUiBinder.class);
+
+    @Inject
+    public AppStepResourcesViewImpl(DEProperties deProperties,
+                                    AppTemplateWizardAppearance appearance,
+                                    AppTemplateAutoBeanFactory factory,
+                                    @Assisted AppTemplateStepLimits limits) {
+        this.appearance = appearance;
+
+        minDiskSpaceEditor = createDataSizeComboBox();
+        memory = createDataSizeComboBox();
+        minCPUCoresEditor = createDoubleComboBox();
+
+        final Long minDiskSpace = limits.getMinDiskSpace();
+        final long diskSpaceEditorMin = (minDiskSpace != null && minDiskSpace > 0) ? minDiskSpace : ONE_GB;
+        buildResourceSizeLimitList(minDiskSpaceEditor, ONE_GB, diskSpaceEditorMin, deProperties.getToolsMaxDiskLimit());
+
+        final Long memoryLimit = limits.getMemoryLimit();
+        final Long minMemoryLimit = limits.getMinMemoryLimit();
+        final long memoryEditorMin = minMemoryLimit != null ? minMemoryLimit : 2 * ONE_GB;
+        final long memoryEditorMax =
+                (memoryLimit != null && memoryLimit > 0) ? memoryLimit : deProperties.getToolsMaxMemLimit();
+        buildResourceSizeLimitList(memory, 2 * ONE_GB, memoryEditorMin, memoryEditorMax);
+
+        final Double maxCPUCores = limits.getMaxCPUCores();
+        final Double minCPUCores = limits.getMinCPUCores();
+
+        final double cpuEditorMin = (minCPUCores != null && minCPUCores > 1) ? Math.round(minCPUCores) : 1;
+        final double cpuEditorMax =
+                (maxCPUCores != null && maxCPUCores > 0) ? Math.round(maxCPUCores) : deProperties.getToolsMaxCPULimit();
```

Comment:
This could be possible even if these checks were the same.

If a tool admin sets min CPUs to 2 and max CPUs to 1 (Belphegor still allows admins to set any arbitrary decimal for min CPUs, but the max CPUs are currently restricted to whole numbers starting with `1`), then in this launch dialog, min CPUs would be set to `2` by default, but only `0` would be available in the drop-down.

So we're counting on tool admins to set realistic min/max values, but `0` will always be available to clear the resource request in case an admin makes a mistake.

Although now I don't remember why I thought I needed to check for `min > 1` here. It must have been while I was debugging with VICE tools that have `0.1` min CPUs set, but checking for `min > 0` doesn't seem to be an issue with the current logic in `buildResourceDoubleLimitList` 🤔 

In fact, I don't think I even need to round the min/max limits passed to this method anymore. I'll update these lines before merging, to get rid of the rounding, and to make both checks the same, which I think makes the logic clearer.

Reply:
Hm... we could probably add a validation check to make sure on the Belphy side that the max is greater than the min.  Or I guess 0 would be a valid max too, so 0 or greater I guess?

Reply:
The last commit of this PR adds these validation checks 👍